huaweicloud/services/rds/resource_huaweicloud_rds_cross_region_backup_strategy.go

```diff
@@ -0,0 +1,328 @@
+		d.Set("backup_type", utils.PathSearch("backup_type", backupStrategies[0], nil)),
+		d.Set("keep_days", utils.PathSearch("keep_days", backupStrategies[0], nil)),
+		d.Set("destination_region", utils.PathSearch("destination_region", backupStrategies[0], nil)),
+		d.Set("destination_project_id", utils.PathSearch("destination_project_id", backupStrategies[0], nil)),
+	)
+
+	if len(backupStrategies) == 1 {
+		mErr = multierror.Append(mErr, d.Set("backup_type", "auto"))
+	} else {
+		mErr = multierror.Append(mErr, d.Set("backup_type", "all"))
+	}
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceBackupStrategyUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	updateBackupStrategyChanges := []string{
+		"backup_type",
+		"keep_days",
+	}
+
+	if d.HasChanges(updateBackupStrategyChanges...) {
+		// updateBackupStrategy: update a RDS cross-region backup strategy.
+		var (
+			updateBackupStrategyProduct = "rds"
+		)
+		updateBackupStrategyClient, err := cfg.NewServiceClient(updateBackupStrategyProduct, region)
+		if err != nil {
+			return diag.Errorf("error creating RDS client: %s", err)
+		}
+
+		if d.HasChange("keep_days") {
+			err = updateCrossRegionBackupStrategy(ctx, updateBackupStrategyClient, d.Timeout(schema.TimeoutUpdate),
+				d.Get("instance_id").(string), buildUpdateBackupStrategyKeepDaysBodyParams(d))
+			if err != nil {
+				return diag.Errorf("error updating RDS cross region backup strategy: %s", err)
+			}
+		}
+		// If backup_type is changed from `all` to `auto`, we should close the incremental backup
+		if d.HasChange("backup_type") {
+			err = updateCrossRegionBackupStrategy(ctx, updateBackupStrategyClient, d.Timeout(schema.TimeoutUpdate),
+				d.Get("instance_id").(string), buildUpdateBackupStrategyBackupTypeBodyParams(d))
+			if err != nil {
+				return diag.Errorf("error updating RDS cross region backup strategy: %s", err)
+			}
+		}
+	}
+	return resourceBackupStrategyRead(ctx, d, meta)
+}
+
+func buildUpdateBackupStrategyKeepDaysBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"policy_para": map[string]interface{}{
+			"backup_type":            utils.ValueIngoreEmpty(d.Get("backup_type")),
+			"keep_days":              utils.ValueIngoreEmpty(d.Get("keep_days")),
+			"destination_region":     utils.ValueIngoreEmpty(d.Get("destination_region")),
+			"destination_project_id": utils.ValueIngoreEmpty(d.Get("destination_project_id")),
+		},
+	}
+	return bodyParams
+}
+
+func buildUpdateBackupStrategyBackupTypeBodyParams(d *schema.ResourceData) map[string]interface{} {
+	params := map[string]interface{}{
+		"backup_type":            "incremental",
+		"destination_region":     utils.ValueIngoreEmpty(d.Get("destination_region")),
+		"destination_project_id": utils.ValueIngoreEmpty(d.Get("destination_project_id")),
+	}
+	if d.Get("backup_type") == "auto" {
```

Comment:
it's better to add some comments to explain why we should set keep_days to 0.

Reply:
done